components/playlist/browser/resources/components/playerControls.tsx

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+import Icon from '@brave/leo/react/icon'
+import styled from 'styled-components'
+import { color } from '@brave/leo/tokens/css'
+
+interface Props {
+  videoElementRef: React.RefObject<HTMLVideoElement>
+  className?: string
+}
+
+const Container = styled.div`
+  display: grid;
+  grid-template-columns: auto auto;
+  justify-content: space-between;
+
+  & > div {
+    display: flex;
+    gap: 16px;
+  }
+`
+
+const StyledIcon = styled(Icon)`
+  color: ${color.icon.default};
+  cursor: pointer;
+`
+
+function Control ({
+  iconName,
+  onClick
+}: {
+  iconName: string
+  onClick: () => void
+}) {
+  return (
+    <div onClick={onClick}>
+      <StyledIcon name={iconName}></StyledIcon>
+    </div>
+  )
+}
+
+export default function PlayerControls ({ videoElementRef, className }: Props) {
+  const [isPlaying, setPlaying] = React.useState(false)
+
+  React.useEffect(() => {
+    const videoElement = videoElementRef.current
+    if (videoElement) {
+      const togglePlayingState = () => {
+        videoElement.paused ? videoElement.play() : videoElement.pause()
+      }
+      const notifyPlaying = () => setPlaying(true)
+      const notifyPaused = () => setPlaying(false)
+
+      videoElement.addEventListener('click', togglePlayingState)
+      videoElement.addEventListener('playing', notifyPlaying)
+      videoElement.addEventListener('pause', notifyPaused)
+      videoElement.addEventListener('ended', notifyPaused)
+      return () => {
+        videoElement.removeEventListener('click', togglePlayingState)
+        videoElement.removeEventListener('playing', notifyPlaying)
+        videoElement.removeEventListener('pause', notifyPaused)
+        videoElement.removeEventListener('ended', notifyPaused)
+      }
+    }
+
+    return () => {}
+  }, [videoElementRef.current])
```

Comment:
Great! I think I don't know the `ref` well. Need to spend sometime for it 😄

Reply:
Yeah, it can be pretty confusing - basically think of refs as an instance variable on an old class style component:

```tsx
class MyComponent {
    foo: HTMLDivElement;

    render() {
        return <div ref={foo}/>
    }
}
```

setting foo to the ref doesn't result in a rerender because there's no call to `setState`. `useRef` is basically instance variables for functions :laughing: (but way more confusing, with lots more footguns).